31_運用監視の環境構築/CFn_Watch_WAF.yaml

```
@@ -0,0 +1,538 @@
+        log_output: "FILE"
+      Tags:
+        - Key: Name
+          Value: !Sub "${EnvType}-rds-parameter-group"
+
+  # SNSトピックの作成
+  AlarmNotificationTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: "Watch-WAF-Topic"
+      Subscription:
+        - Protocol: email
+          Endpoint: !Ref NotificationEmail
+      Tags:
+        - Key: Name
+          Value: !Sub "${EnvType}-alarm-topic"
+
+  # SNSトピックポリシーの作成
+  SNSTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    Properties:
+      Topics:
+        - !Ref AlarmNotificationTopic
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Sid: AllowCloudWatchAlarms
+            Effect: Allow
+            Principal:
+              Service: cloudwatch.amazonaws.com
+            Action: sns:Publish
+            Resource: !Ref AlarmNotificationTopic
+
+  # WAFログ設定
+  WAFLogging:
+    Type: AWS::WAFv2::LoggingConfiguration
+    DependsOn: 
+      - WebACL
+      - WAFLogGroup
+    Properties:
+      ResourceArn: !GetAtt WebACL.Arn
+      LogDestinationConfigs:
+        - !GetAtt WAFLogGroup.Arn
+      RedactedFields:
+        - Method: {}
+        - QueryString: {}
+        - SingleHeader:
+            Name: Authorization
+
+  # IP許可用IPセット
+  AllowIPSet:
+    Type: AWS::WAFv2::IPSet
+    Properties:
+      Name: "Allow-IP-Set"
+      Description: Allowed IP address ranges
+      Scope: REGIONAL
+      IPAddressVersion: IPV4
+      Addresses: 
+        - "150.9.190.229/32"  # 実際の許可IPに置き換え
+
+  # WAF WebACLの作成
+  WebACL:
+    Type: AWS::WAFv2::WebACL
+    Properties:
+      Name: Watch-WAF
+      Scope: REGIONAL
+      Description: WAF for Watch-WAF-alb
+      DefaultAction:
+        Block: {}
+      VisibilityConfig:
+        CloudWatchMetricsEnabled: true
+        MetricName: Watch-WAF
+        SampledRequestsEnabled: true
+      Rules:
+        - Name: AllowSpecificIP
+          Priority: 0
+          Action:
+            Allow: {}
+          Statement:
+            IPSetReferenceStatement:
+              Arn: !GetAtt AllowIPSet.Arn
+          VisibilityConfig:
+            CloudWatchMetricsEnabled: true
+            MetricName: AllowedByIP
+            SampledRequestsEnabled: true
+
+  # WAFログ保存用CloudWatch Logsグループの作成
+  WAFLogGroup:
+    Type: AWS::Logs::LogGroup
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Retain
+    Properties:
+      LogGroupName: "aws-waf-logs-Watch-WAF"
+      RetentionInDays: 30
+      Tags:
+        - Key: Name
+          Value: !Sub "${EnvType}-waf-log-group"
+
+  # ALBへWAFを関連付け
+  WebACLAssociation:
+    Type: AWS::WAFv2::WebACLAssociation
+    DependsOn: 
+      - ALB
+      - WebACL
+    Properties:
+      ResourceArn: !GetAtt ALB.LoadBalancerArn
+      WebACLArn: !GetAtt WebACL.Arn
+
+  # CloudWatchアラームの設定
+  WAFBlockedRequestsAlarm:
+    Type: AWS::CloudWatch::Alarm
+    Properties:
+      AlarmName: "Watch-WAF-blocked-requests-alarm"
+      AlarmDescription: "許可されていないIPからのアクセスを検知"
```

意味は同じですが、以下の方がルールの内容とマッチするので個人的には好きです。
```suggestion
      AlarmDescription: "許可しているIP以外からのアクセスを検知"
```

---

ありがとうございます！
同じように考えられるようにしていきます！
修正いたします！